homeassistant/components/swidget/config_flow.py

```diff
@@ -0,0 +1,92 @@
+"""Config flow for swidget integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from swidget.exceptions import SwidgetException
+from swidget.swidgetdevice import SwidgetDevice
+import voluptuous as vol  # type: ignore[import-untyped]
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DISCOVERY_INTERVAL = timedelta(minutes=15)
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, str]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    device: SwidgetDevice = SwidgetDevice(
+        host=data["host"],
+        token_name="x-secret-key",
+        secret_key=data.get("password"),
+        use_https=True,
+        use_websockets=False,
+    )
+    try:
+        await device.update()
+        await device.stop()
+    except SwidgetException as exc:
+        raise CannotConnect from exc
+    friendly_name: str = device.friendly_name
+    unique_id: str = f"{device.id}_{device.mac_address}"
+    return {"title": friendly_name, "unique_id": unique_id}
```

Comment:
```suggestion
    return {"title": device.friendly_name, "unique_id": f"{device.id}_{device.mac_address}"}
```
since you're not reusing those, it's a bit more readable imo

Reply:
Fixed